tokio-util/src/sync/tests/loom_cancellation_token.rs

```diff
@@ -174,3 +174,46 @@ fn cancel_parent_and_child() {
         assert_ok!(th3.join());
     });
 }
+
+#[test]
+fn run_until_cancelled_completes() {
+    loom::model(|| {
+        block_on(async {
+            let token = CancellationToken::new();
+
+            let fut = async {
+                tokio::time::sleep(std::time::Duration::from_millis(500)).await;
+                42
+            };
+
+            if let Some(res) = token.run_until_cancelled(fut).await {
+                assert_eq!(res, 42);
+            } else {
+                panic!("Should not happen since we are not cancelling the token");
+            }
+        });
+    });
+}
+
+#[test]
+fn run_until_cancelled_with_cancel() {
+    loom::model(|| {
+        block_on(async {
+            let token = CancellationToken::new();
+            let token1 = token.clone();
+
+            let th1 = std::thread::spawn(move || {
+                std::thread::sleep(std::time::Duration::from_millis(500));
+                token1.cancel();
+            });
```

Comment:
We do not use sleeping in tests. We have 845 tests right now. Imagine how long it would take to run the tests if every single one had a 0.5 second sleep.

Reply:
Yeah that was a stupid idea. I changed the test to use a channel instead of a sleep.